rewrite-core/src/main/java/org/openrewrite/NpmExecutionContextView.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite;
```

Comment:
I wonder if `rewrite-core`, and package `org.openrewrite` are the best place for this NPM specific context. We seem to be spreading our frontend related classes across `rewrite-javascript`, `rewrite-nodejs`, `rewrite-codemods` and `rewrite-codemods-ng`. Adding `rewrite-core` to that list for something NPM specific might then further create confusion as to what goes where. Would it make sense to move the NPM classes elsewhere, and only keep the `ShellExecutor` in rewrite-core?

Reply:
Yes, we'll move the npm specific code to rewrite-nodejs